kernel/packages/shared/social/hotScenes.ts

```diff
@@ -0,0 +1,123 @@
+import { ParcelArray } from 'shared/comms/interface/utils'
+import { Candidate } from 'shared/dao/types'
+import { StoreContainer } from 'shared/store/rootTypes'
+import { fetchSceneIds } from 'decentraland-loader/lifecycle/utils/fetchSceneIds'
+import { fetchSceneJson } from 'decentraland-loader/lifecycle/utils/fetchSceneJson'
+import { SceneJsonData } from 'shared/types'
+import { getSceneNameFromAtlasState, postProcessSceneName } from 'shared/atlas/selectors'
+
+declare const globalThis: StoreContainer
+
+type CrowdedSceneRealmInfo = {
+  realm: {
+    serverName: string
+    layerName: string
+  }
+  usersCount: number
+  maxUsers: number
+}
+
+type CrowdedSceneInfo = {
+  name: string
+  coordinates: {
+    x: number
+    y: number
+  }
+  realmsInfo: CrowdedSceneRealmInfo[]
+}
+
+type CandidateCrowdedScene = {
+  id: string
+  scene: SceneJsonData | undefined
+  baseCoord: ParcelArray
+  usersCount: number
+}
+
+export async function fetchHotScenes(): Promise<CrowdedSceneInfo[]> {
+  const candidates = await refreshCandidatesStatuses()
+
+  let crowdedScenes: Record<string, CrowdedSceneInfo> = {}
+
+  const filteredCandidates = candidates.filter(
+    (candidate) => candidate.layer && candidate.layer.usersCount > 0 && candidate.layer.usersParcels
+  )
+
+  for (const candidate of filteredCandidates) {
+    const candidateScenes = await getCrowdedScenesFromCandidate(candidate)
+
+    candidateScenes.forEach((sceneInfo) => {
+      if (!crowdedScenes[sceneInfo.id]) {
+        crowdedScenes[sceneInfo.id] = crowdedSceneInfoFromCandidateScene(sceneInfo)
+      }
+      crowdedScenes[sceneInfo.id].realmsInfo.push({
+        realm: {
+          serverName: candidate.catalystName,
+          layerName: candidate.layer.name
+        },
+        maxUsers: candidate.layer.maxUsers,
+        usersCount: sceneInfo.usersCount
+      })
+    })
+  }
+
+  const sceneValues = Object.values(crowdedScenes)
+  sceneValues.forEach((scene) => scene.realmsInfo.sort((a, b) => (a.usersCount > b.usersCount ? -1 : 1)))
+
+  return sceneValues.sort((a, b) => (countUsers(a) > countUsers(b) ? -1 : 1))
+}
+
+function countUsers(a: CrowdedSceneInfo) {
+  return a.realmsInfo.reduce((total, realmInfo) => total + realmInfo.usersCount, 0)
+}
+
+function createCandidateCrowdedScene(
+  id: string,
+  baseCoord: string,
+  sceneJsonData: SceneJsonData | undefined
+): CandidateCrowdedScene {
+  return {
+    id,
+    baseCoord: baseCoord.split(',').map((str) => parseInt(str, 10)) as ParcelArray,
+    scene: sceneJsonData,
+    usersCount: 1
+  }
+}
+
+async function getCrowdedScenesFromCandidate(candidate: Candidate): Promise<CandidateCrowdedScene[]> {
+  let scenes: Record<string, CandidateCrowdedScene> = {}
+
+  const tiles = candidate.layer.usersParcels?.map((value) => `${value[0] | 0},${value[1] | 0}`) ?? []
+
+  for (const tile of tiles) {
+    let id = (await fetchSceneIds([tile]))[0]
+    let land = null
+
+    if (id) {
+      land = (await fetchSceneJson([id]))[0]
```

Comment:
Might prove better to fetch both ids and data for the whole set of tiles before iterating though the collection.